db/db.go

```diff
@@ -88,7 +148,10 @@ func NewLoader(
 		moduleMismatchMode:       moduleMismatchMode,
 		logger:                   logger,
 		tracer:                   tracer,
+		maxParallelFlushes:       maxParallelFlushes,
 	}
+	l.mu = sync.Mutex{}
```

Comment:
you don't need to store the mutex, you can just do

`l.cond = sync.NewCond(&sync.Mutex{})`

and then access it using 

`l.cond.L.Lock()`